### PR TITLE
nested exception tests for discussion

### PR DESCRIPTION
**Pull request by jjb**
see discussion in https://github.com/ruby/timeout/issues/38

### CODE REVIEW

File: test/test_timeout.rb [hunk 1]
```diff
@@ -63,6 +63,42 @@ def test_nested_timeout
...
+      Timeout.timeout(0.1) {
+        Timeout.timeout(1) {
+          raise Timeout::ExitException.new("inner message")
+        }
+      }
+    rescue Exception => e
+      raised_exception = e
+    end
+
+    assert_equal 'inner message', e.message
```
Comment (eregon) [inline test/test_timeout.rb:99]: e.message -> raised_exception.message

Comment (jjb) [reply test/test_timeout.rb:99]: thanks @eregon - are you saying that that is the more important thing to test?

Comment (jjb) [reply test/test_timeout.rb:99]: added the additional check https://github.com/ruby/timeout/pull/43/files

**Comment by eregon** [comment]
Yeah, this is the current and what looks to be the correct behavior, so good to be merged.

**Pull request merged**